kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/BaseContributor.java

```diff
@@ -38,6 +38,9 @@ public Class<? extends BaseConfig> getConfigType(String shortName) {
     @Override
     public T getInstance(String shortName, BaseConfig config) {
         InstanceBuilder<? extends BaseConfig, T> instanceBuilder = shortNameToInstanceBuilder.get(shortName);
+        if (requiresConfig(shortName) && config == null) {
```

Comment:
I think `BaseContributor` should support this so that users can do something along the lines of:

```
        public <T extends BaseConfig> BaseContributorBuilder<L> add(String shortName, Class<T> configClass, Function<T, L> instanceFunction, boolean requiresConfig) {
```

used like:

```
BaseContributorBuilder<Filter> FILTERS = BaseContributor.<Filter> builder()
            .add("ProduceRequestTransformation", ProduceRequestTransformationConfig.class, ProduceRequestTransformationFilter::new, false)
            .add("FetchResponseTransformation", FetchResponseTransformationConfig.class, FetchResponseTransformationFilter::new, true)
            ```

Reply:
could be delayed to another PR though

Reply:
I like it, its a nicer way of doing things than random if blocks in `requiresConfig`